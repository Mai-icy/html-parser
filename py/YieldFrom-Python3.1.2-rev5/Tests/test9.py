#
#   Test handling exception while delegating 'close'
#

def g1():
	try:
		print("Starting g1")
		yield "g1 ham"
		yield from g2()
		yield "g1 eggs"
	finally:
		print("Finishing g1")

def g2():
	try:
		print("Starting g2")
		yield "g2 spam"
		yield "g2 more spam"
	finally:
		print("Finishing g2")
		raise ValueError("nybbles have exploded with delight")

g = g1()
for i in range(2):
	x = next(g)
	print("Yielded", x)
g.close()
