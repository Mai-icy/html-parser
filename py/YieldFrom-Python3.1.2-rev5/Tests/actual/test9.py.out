Starting g1
Yielded g1 ham
Starting g2
Yielded g2 spam
Finishing g2
Finishing g1
Traceback (most recent call last):
  File "test9.py", line 17, in g2
    yield "g2 spam"
GeneratorExit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test9.py", line 27, in <module>
    g.close()
  File "test9.py", line 9, in g1
    yield from g2()
  File "test9.py", line 21, in g2
    raise ValueError("nybbles have exploded with delight")
ValueError: nybbles have exploded with delight
